test/e2e/search.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { test, expect } from "@playwright/test";
+import { home_url } from "../helper-config";
+import { SearchPage } from "../pages/SearchPage";
+
+const existingBook: string = "Luffy";
+const nonexistingBook: string = "hfur";
+
+test.beforeEach(async ({ page }) => {
+  await page.goto(home_url);
+});
+
+test("search existing book", async ({ page }) => {
+  const searchPage = new SearchPage({ page });
+  await searchPage.search(existingBook);
+  await expect(page.locator(".results")).toContainText(existingBook);
```

Comment:
locators should not be used in the tests, because if they change you will have to change multiple places.
create a function in the page object to get the results. e.g. 
```
searchPage.getResults()
```
and then you can do the assertion here